misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/RangeFormatterTest.java

```diff
@@ -38,46 +44,25 @@
 /**
  * The abstract class that is extended by all range formatting test classes.
  *
- * @since 1.2.10
+ * @since 2.0.0
  */
-public abstract class RangeFormatter {
+public abstract class RangeFormatterTest {
     private final Path resourceDirectory = Paths.get("src").resolve("test").resolve("resources").toAbsolutePath();
-    private Path buildDirectory = Paths.get("build").toAbsolutePath().normalize();
     private static final String ASSERT_DIR = "assert";
     private static final String SOURCE_DIR = "source";
 
-    @Test(dataProvider = "test-file-provider")
-    public void test(String source, int[][] positions)
-            throws IOException, FormatterException {
-        Path assertFilePath = Paths.get(resourceDirectory.toString(), this.getTestResourceDir(), ASSERT_DIR, source);
-        Path sourceFilePath = Paths.get(resourceDirectory.toString(), this.getTestResourceDir(), SOURCE_DIR, source);
+    private static final Gson gson = new Gson();
 
-        String content = getSourceText(sourceFilePath);
+    @Test(dataProvider = "test-file-provider")
+    public void test(Path sourceFilePath, Path assertFilePath, ArrayList<LineRange> lineRanges) throws IOException,
+            FormatterException {
+        String content = getFileContent(sourceFilePath);
         TextDocument textDocument = TextDocuments.from(content);
         SyntaxTree syntaxTree = SyntaxTree.from(textDocument, sourceFilePath.toString());
```

Comment:
Is there a reason for using this overloaded `SyntaxTree.from` method?

Reply:
It was used as the previous approach required to access the positions of the syntax tree. 
The revised approach no longer requires the overloaded method call. 
Fixed it.